apps/shipments/views.py

```diff
@@ -206,3 +207,22 @@ def create(self, request, *args, **kwargs):
         permission_link = PermissionLink.objects.create(**serializer.validated_data)
 
         return Response(PermissionLinkSerializer(permission_link).data, status=status.HTTP_201_CREATED)
+
+
+class EmailShipmentPermissionLink(viewsets.ViewSet):
+    http_method_names = ['post', ]
+    permission_classes = ((IsShipmentOwner,) if settings.PROFILES_ENABLED else (permissions.AllowAny,))
+
+    def create(self, request, *args, **kwargs):
+        serialize = EmailShipmentDetailsSerializer(data=request.data, context={'shipment_id': kwargs['shipment_pk'],
+                                                                               'user': request.user})
+        serialize.is_valid(raise_exception=True)
+        email = serialize.validated_data['email']
+        email_context = serialize.get_context
```

Comment:
I think in this case it's going to be cleaner to just fetch the username/link/subject within the view than have a Serializer.get_context property.